### DISCUSSION TITLE
Uncaught Exception question

### DISCUSSION THREAD

**Discussion by alnospam64** [tsightler/ring-mqtt#325] [Support Q&A]
Hi,
 I am running Openhab 3.2 on a Raspberry 4 
I have managed to get everything installed (standard instal) and if I run the js file from the opt/ring-mqtt folder all is fine

  ring-mqtt Using configuration file: ./config.json +0ms
  ring-mqtt Attempting connection to Ring API using refresh token from file: ./config.json +454ms
  ring-mqtt Connection to Ring API successful +1s
  ring-mqtt Starting connection to MQTT broker... +16ms
  ring-mqtt Config file saved with updated refresh token. +256ms
  ring-mqtt MQTT connection established, processing locations... +29ms
  ring-mqtt -------------------------------------------------------------------------------- +2s
  ring-mqtt New location: xxxxx(xxxxxxx) +1ms
  ring-mqtt -------------------------------------------------------------------------------- +2ms
  ring-mqtt true +0ms
  ring-mqtt -------------------------------------------------------------------------------- +5ms
  ring-mqtt   New device: Front Door (9c7613395c5c) +11ms
  ring-mqtt               doorbell_graham_cracker +0ms
  ring-mqtt   New device: Hallway (9c76132b7d63) +3ms
  ring-mqtt               chime_v2 +1ms
  ring-mqtt -------------------------------------------------------------------------------- +0ms
  ring-mqtt Ring location/device data updated, sleeping for 5 seconds. +1ms
 ring-mqtt [Front Door] ring/xxxxxx/camera/9c7613395c5c/status online +0ms

but if I let the service run I get this error

2022-02-21T10:22:00.259Z ring-mqtt Ring location/device data updated, sleeping for 5 seconds.
2022-02-21T10:22:02.295Z ring-mqtt ERROR - Uncaught Exception
Error: spawn rtsp-simple-server ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:274:19)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn rtsp-simple-server',
  path: 'rtsp-simple-server',
  spawnargs: [ '/opt/ring-mqtt/lib/../config/rtsp-simple-server.yml' ]

Any idea of what is going on?

**Comment by tsightler** [comment]
The second pre-requisite in the standard install document notes that rtsp-simple-server must be available in the system path.  The "ENOENT" error means "File not found" so it means this requirement has not been met.  Where did you install rtsp-simple-server and do you have that location in the path for the services file?

You might want to consider a Docker install which takes care of all of this for you.

**Reply by alnospam64** [reply]
Hi, thanks for the quick reply

I have installed the rtsp-simple-server  in /usr/local/bin/rtsp-simple-server
and added 
export PATH=$PATH:/usr/local/bin/rtsp-simple-server
in my .bashrc and .bash_profile

I will try the docker method but I am confused on why it would work if I did run the js file manually

**Reply by tsightler** [reply]
Assuming you are using the example ring-mqtt.service file included in this repo then it's because that file does not include /usr/local/bin in the path so you'll need to modify the file to include it.  Services don't generally run under user context so adding to .bashrc/.bash_profile are not really useful for that case.  You can look on line 11 to see where you need to modify the path.

**Comment by alnospam64** [comment]
Thanks. Now I have a13 error EACCES.
I am going to try the docker route... Thanks

**Reply by tsightler** [reply]
Clearly a permissions related issues, but I'm not sure how that's possible since the example ring-mqtt.service file is setup to use root.  If you've changed it to another user then you'll need to make sure permissions are correct on the /opt/ring-mqtt directory and for the rtsp-simple-server (maybe you didn't set execute bit?).

Good luck!
